### PR TITLE
issue 6: WantToHelp added

### PR DESCRIPTION
**Pull request by OtchereDev**
#6 please i have completed the issue

### CODE REVIEW

**Comment by OtchereDev** [comment]
#4 i have also solved this issue

File: src/administration/models.py [hunk 1]
```diff
@@ -112,3 +113,23 @@ class Meta:
...
+
+
+class OpportunityCategory(models.Model):
+    name=models.CharField(max_length=225)
+    slug=models.SlugField()
+    opportunities=models.ManyToManyField(Opportunity)
+    created=models.DateTimeField( auto_now_add=True)
+    updated=models.DateTimeField( auto_now=True)
+
+    def save(self, *args, **kwargs):
```
Comment (danpercic86) [inline src/administration/models.py]: it's not necessary to slugify on save, slug must be required and therefore can't be null

File: src/administration/models.py [hunk 2]
```diff
@@ -112,3 +113,23 @@ class Meta:
...
+    email=models.EmailField (max_length=255)
+    description=models.TextField()
+
+
+
+class OpportunityCategory(models.Model):
+    name=models.CharField(max_length=225)
+    slug=models.SlugField()
+    opportunities=models.ManyToManyField(Opportunity)
+    created=models.DateTimeField( auto_now_add=True)
```
Comment (danpercic86) [inline src/administration/models.py]: remove before auto_now_add and auto_add

**Review by danpercic86 — CHANGES_REQUESTED** [review]
Add WantToHelpViewSet and OpportunityCatViewSet to url's router

File: src/administration/views.py [hunk 3]
```diff
@@ -39,3 +40,15 @@ class ArticleViewSet(ReadOnlyModelViewSet):
 class NewsletterViewSet(ReadOnlyModelViewSet):
     serializer_class = NewsletterSerializer
     queryset = Newsletter.objects.filter(is_published=True)
+
+
+class WantToHelpViewSet(generics.ListCreateAPIView):
```
Comment (danpercic86) [inline src/administration/views.py]: there will be only one WantToHelp added at a time. ListCreateAPIView it's not necessary. Use APIView/ViewSet that allows adding only one entity

File: src/administration/views.py [hunk 4]
```diff
@@ -39,3 +40,15 @@ class ArticleViewSet(ReadOnlyModelViewSet):
...
+    queryset = WantToHelp.objects.all()
+    serializer_class = WantToHelpSerializer
+
+
+class OpportunityCatViewSet(ReadOnlyModelViewSet):
+    serializer_class = OpportunityCatSerializer
+    queryset = OpportunityCategory.objects.filter(is_published=True)
```
Comment (danpercic86) [inline src/administration/views.py]: OpportunityCategory does not have "is_published" attribute

File: src/administration/serializers.py [hunk 5]
```diff
@@ -43,3 +43,19 @@ class NewsletterSerializer(serializers.ModelSerializer):
...
+        model=WantToHelp
+        fieldss='__all__'
+
+
+class OpportunityCatSerializer(serializers.ModelSerializer):
+    class Meta:
+        model=OpportunityCategory
+        fields=['name',
+                'slug',
+                'opportunities']
```
Comment (danpercic86) [inline src/administration/serializers.py]: maybe you should use depth attribute to get opportunities fields

File: src/administration/models.py [hunk 6]
```diff
@@ -112,3 +113,23 @@ class Meta:
 
     def __str__(self):
         return self.email
+
+class WantToHelp(models.Model):
+    name=models.CharField(max_length=225)
+    email=models.EmailField (max_length=255)
+    description=models.TextField()
+
+
```
Comment (danpercic86) [inline src/administration/models.py:122]: there should be only 2 blank lines

File: src/administration/admin.py [hunk 7]
```diff
@@ -72,6 +72,18 @@ class NewsletterAdmin(admin.ModelAdmin):
     prepopulated_fields = {'slug': ('email',)}
 
 
+class WantToHelpAdmin(admin.ModelAdmin):
+    list_display = ('name', 'email')
+    list_filter = ('name', 'email')
+    search_fields = ('email',)
+
```
Comment (danpercic86) [inline src/administration/admin.py:79]: there should be 2 blank lines

**Comment by OtchereDev** [comment]
Hello @danpercic86 the changes have been resolved, review and let me know if they are okay

**Pull request merged**